app/components/article-content.js

```diff
@@ -629,14 +624,28 @@ export default Component.extend(
             return;
           }
 
+          // keep here for tracking purposes.
+          // We want to know if we have targeting but no space for the unit
+          if (!h2Elements[indexForUnit]) {
+            trackAffiliateUnit(unit, {
+              category: 'mercury-affiliate_incontent_recommend',
+              label: 'affiliate_not_shown',
+              action: 'no-impression',
+            });
+            return;
+          }
+
           const unitPlaceholder = document.createElement('div');
           const unitWrapper = document.createElement('div');
           unitWrapper.appendChild(unitPlaceholder);
           h2Elements[indexForUnit].insertAdjacentElement('beforebegin', unitWrapper);
 
           this.renderedComponents.push(this.renderComponent({
             name: 'affiliate-unit',
-            attrs: unit,
+            attrs: {
+              unit,
```

Comment:
Doesn't this change the structure of the attributes? Instead of `attrs` being set to the value of `unit`, the unit object is now a property, so the properties of `unit` are one level lower in the attributes object.

Reply:
Yup, I modified everything that followed from this change.

Reply:
The only change I had to make was https://github.com/Wikia/mobile-wiki/pull/1865/files#diff-db0f68f01afcb65f9f6ceb052a87d8e6

Reply:
Yup, that's what @JoshuaRogan did - `affiliate-unit` component now accents `unit` prop instead of unit's attributes.